package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,652 @@
+
+        Parameters
+        ----------
+        fname : str
+            The filename to open.
+        mode : ('r', 'w')
+            The mode in which to open the file, either 'r' read or 'w' write
+
+        Raises
+        ------
+        IOError
+        """
+        if self.is_open:
+            self.close()
+
+        if mode == 'r':
+            fio_mode = FIO_READ
+        elif mode == 'w':
+            fio_mode = FIO_WRITE
+        else:
+            raise IOError("unkown mode '{}', use either r or w".format(mode))
+        self.mode = str(mode)
+
+        ok = fio_open(self.fname, fio_mode, <fio_fd*> &self.fp)
+        if ok != 0:
+            raise IOError("couldn't open file: {}\n"
+                          "ErrorCode: {}".format(self.fname, ok))
+        self.is_open = True
+        self.current_frame = 0
+        self.reached_eof = False
+        self.wrote_header = False
+        # Has to come last since it checks the reached_eof flag
+        if self.mode == 'r':
+            self._read_header()
+
+    def close(self):
+        """Close the open DCD file
+
+        Raises
+        ------
+        IOError
+        """
+        if self.is_open:
+            # In case there are fixed atoms we should free the memory again.
+            # Both pointers are guaranted to be non NULL if either one is.
+            if self.freeind != NULL:
+                close_dcd_read(self.freeind, self.fixedcoords);
+
+            ok = fio_fclose(self.fp)
+            self.is_open = False
+            if ok != 0:
+                raise IOError("couldn't close file: {}\n"
+                              "ErrorCode: {}".format(self.fname, ok))
+
+
+    cdef void _read_header(self):
+        """read header and populate internal fields"""
+        if not self.is_open:
+            raise IOError("No file open")
+
+        cdef char* c_remarks
+        cdef int len_remarks = 0
+        cdef int nsets
+
+        ok = read_dcdheader(self.fp, &self.natoms, &nsets, &self.istart,
+                            &self.nsavc, &self.delta, &self.nfixed, &self.freeind,
+                            &self.fixedcoords, &self.reverse_endian,
+                            &self.charmm, &c_remarks, &len_remarks)
+        if ok != 0:
+            raise IOError("Reading DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+        if c_remarks != NULL:
+            py_remarks = <bytes> c_remarks[:len_remarks]
+            free(c_remarks)
+        else:
+            py_remarks = ""
+
+        self.ndims = 3 if not self.charmm & DCD_HAS_4DIMS else 4
+        self.n_frames = self._estimate_n_frames()
+        self.b_read_header = True
+
+        # make sure fixed atoms have been read
+        try:
+            self.read()
+            self.seek(0)
+        except:
+            # if this fails the file is empty. Set flag and warn using during read
+            if self.n_frames != 0:
+                raise IOError("DCD is corrupted")
+
+        if sys.version_info[0] < 3:
```

Comment:
I can have a look again. But I think six tries to standardize it to utf-8 while we here explicitly use ascii.